docker-compose/testnet/docker-compose.yml

```diff
@@ -4,8 +4,9 @@ services:
     container_name: bitcoind-testnet
     build: ../../Dockerfile/bitcoind
     restart: unless-stopped
-    command:
-      - "-rpcuser=bitcoin -rpcpassword=bitcoin"
+    command: "-rpcuser=bitcoin -rpcpassword=bitcoin -testnet \
+              -rpcbind=0.0.0.0 -rpcallowip=172.16.0.0/12 \
+              -rpcallowip=192.168.0.0/16 -rpcallowip=10.0.0.0/8"
```

Comment:
We are not using this config in main net, why do we need it here? I expect this may break with docker updates; AFAIR there is no guarantee that docker will always stick to these IP ranges (especially with 172.16)

Reply:
AFAIK docker always sticks to private addresses, so addresses that are in the ranges `172.16.0.0 – 172.31.255.255`, `192.168.0.0 – 192.168.255.255` and `10.0.0.0 – 10.255.255.255`. the `rpcallowip` option is needed in order to allow RPC requests from those IPs (c-lightning would not work without these options)

moreover the command option doesn't work if specified as a list

I've never tried the mainnet dockerization, are you sure it's working?

Reply:
I am absolutely sure that both mainnet and signet Dockerfiles and docker-composes are all working - I spent two days debugging it and now they are running on both mac Macbook and server

Reply:
that's strange, c-lightning should not be able to connect to bitcoind without the `rpcallowip` option.

are you sure you're running the stack from current `master`?

I should have a mainnet chain somewhere, I'll test the compose in order to confirm your statement (don't trust, verify :wink:)